source/common/upstream/cluster_manager_impl.h

```diff
@@ -361,6 +362,20 @@ class ClusterManagerImpl : public ClusterManager, Logger::Loggable<Logger::Id::u
   // This map is ordered so that config dumping is consistent.
   typedef std::map<std::string, ClusterDataPtr> ClusterMap;
 
+  struct PendingUpdates {
+    Event::TimerPtr timer;
+    std::set<HostSharedPtr> added;
```

Comment:
This hashes on the pointer value I think. Do we have a guarantee that two identical hosts will have the same pointer value from the caller of `scheduleUpdate()`?

Reply:
yeah good point, we should track the host's address instead.